go-controller/pkg/informer/informer_test.go

```diff
@@ -113,8 +113,8 @@ var _ = Describe("Informer Event Handler Tests", func() {
 		_, err := k.CoreV1().Pods(namespace).Create(context.TODO(), pod, metav1.CreateOptions{})
 		Expect(err).NotTo(HaveOccurred())
 
-		Consistently(deletes).Should(Equal(int32(0)), "deletes")
-		Eventually(adds, 3).Should(Equal(int32(1)), "adds")
+		Consistently(atomic.LoadInt32(&deletes)).Should(Equal(int32(0)), "deletes")
+		Eventually(atomic.LoadInt32(&adds), 3).Should(Equal(int32(1)), "adds")
```

Comment:
This code (both original and modified) is not using `Consistently`/`Eventually` correctly; the argument gets evaluated when you first call `Consistently`/`Eventually`, so in both versions, it is only asserting about the initial values of `deletes`/`adds`, not their eventual values. You need to say:

    Consistently(func() int32 { return atomic.LoadInt32(&deletes) }).Should(Equal(int32(0)), "deletes")
    Eventually(func() int32 { return atomic.LoadInt32(&adds) }).Should(Equal(int32(1)), "adds")

although if this is not already flaky, then something must be forcing it to serialize already and we don't need the async versions...

Likewise for some of the other examples modified below, and probably some others nearby as well.

Reply:
TIL, https://medium.com/@william.la.martin/ginkgotchas-yeh-also-gomega-13e39185ec96although

> flaky, then something must be forcing it to serialize already and we don't need the async versions...

it seems is already flaky

> Likewise for some of the other examples modified below, and probably some others nearby as well.

heh, seems that this is incorrect in most part of the repo :shrug: